test/app.test.js

```diff
@@ -72,6 +72,19 @@ describe("app", function() {
 
         });
 
+        describe('Facility_Code_Confirm', function () {
+            it('should got to MSISDN entry', function () {
+                    return tester
+                        .setup.user.state('Facility_Code_Confirm')
+                        .input('1')
+                        .check.interaction({
+                            state: 'MSISDN_Entry',
+                            reply: /Please enter the South African cell phone number/
+                        })
+                        .run();
+            });
+        });
+
```

Comment:
Yes :) Which is why I'm asking you review it :) It's the best way to learn.

Vumi Go's jsbox essentially allows people to design state machines. Each screen is a state and when certain conditions are met it transitions to the next state, which is a new screen (which may end up being USSD, SMS, Twitter tweet etc ...)